src/test/java/bio/terra/service/filedata/flight/ingest/IngestFilePrimaryDataStepTest.java

```diff
@@ -67,8 +68,7 @@ public void setup() {
   public void testDoStepSelfHostedRetryThenSucceed() {
     // Dataset is self-hosted
     when(dataset.isSelfHosted()).thenReturn(true);
-
-    // Step's first run throws retryable error, second run succeeds
+    // Step Run 1 - throws retryable "InvalidUserProjectException" exception
     FSFileInfo fileInfo = mock(FSFileInfo.class);
     when(gcsPdao.linkSelfHostedFile(any(), any(), any()))
         .thenThrow(new InvalidUserProjectException("retryable"))
```

Comment:
I think the `thenReturn` on line 75 is never used.  I am trying to remember why I chained the mock overrides like this, but am drawing a blank.

It may be easier to follow these tests if we either chain all the mock overrides to start…
```
when(gcsPdao.linkSelfHostedFile(any(), any(), any()))
    .thenThrow(…)
    .thenThrow(…)
    .thenReturn(…);
```

…or define the behavior at the top of each step invocation:
```
when(gcsPdao.linkSelfHostedFile(any(), any(), any()))
    .thenThrow(…);
…
when(gcsPdao.linkSelfHostedFile(any(), any(), any()))
    .thenThrow(…);
…
when(gcsPdao.linkSelfHostedFile(any(), any(), any()))
    .thenReturn(…);
```

Reply:
Ah, ok, thanks! I misunderstood what that mock call was doing. I've changed it to be one chained call with the two retryable exceptions and then a successful call.